encrypted-media/scripts/playback-temporary-waitingforkey.js

```diff
@@ -0,0 +1,90 @@
+function runTest(config) {
+
+    var testname = config.keysystem + ', successful playback, temporary, '
+                                    + /video\/([^;]*)/.exec( config.videoType )[ 1 ]
+                                    + ', waitingforkey event';
+
+    var configuration = {   initDataTypes: [ config.initDataType ],
+                            audioCapabilities: [ { contentType: config.audioType } ],
+                            videoCapabilities: [ { contentType: config.videoType } ],
+                            sessionTypes: [ 'temporary' ] };
+
+    async_test( function( test ) {
+
+        var _video = config.video,
+            _mediaKeys,
+            _mediaKeySessions = [ ],
+            _mediaSource;
+
+        function onMessage(event) {
+            assert_any( assert_equals, event.target, _mediaKeySessions );
+            assert_true( event instanceof window.MediaKeyMessageEvent );
+            assert_equals( event.type, 'message');
+
+            assert_any( assert_equals,
+                        event.messageType,
+                        [ 'license-request', 'individualization-request' ] );
+
+            config.messagehandler( config.keysystem, event.messageType, event.message )
+            .then( function( response ) {
+
+                event.target.update( response )
+                .catch(function(error) {
+                    forceTestFailureFromPromise(test, error);
+                });
+            });
+        }
+
+        function onWaitingForKey(event) {
+
+            // Not using waitForEventAndRunStep() to avoid too many
+            // EVENT(onTimeUpdate) logs.
+            _video.addEventListener('timeupdate', onTimeupdate, true);
+
+            // Create sessions now if we do not have them
+            if ( _mediaKeySessions.length === 0 ) {
```

Comment:
The number and timing of events varied between clearkey and widevine on Chrome. I suspect it may be due to timing. Originally, the test fetched the first key and waiting for playback to start before fetching the second. This worked on clearkey, but with Widevine, it wants both keys before it will begin playback. I guess this could be because the media with the second key is so close to the start (we have 6 seconds with key1 followed by 6 seconds with key 2).

I should probably switch to using the single-key content and testing for a single waitingforkey event before playback starts.